anaDACScan.py

```diff
@@ -0,0 +1,288 @@
+        nominal *= pow(10.0,-9)
+    else:
+        print("Error: unexpected units: '%s'"%nominalDacValues[nameX][1])
+        exit(1)
+
+    calInfo = {}
+
+    if args.calFileList != None:
+        for line in open(args.calFileList):
+            if line[0] == "#":
+                continue
+            line = line.strip(' ').strip('\n')
+            first = line.split(' ')[0].strip(' ')
+            second = line.split(' ')[1].strip(' ')
+            tuple_calInfo = parseCalFile(second)
+            calInfo[int(first)] = {'slope' : tuple_calInfo[0], 'intercept' : tuple_calInfo[1]}
+
+    #for each OH, check if calibration files were provided, if not search for the calFile in the $DATA_PATH, if it is not there, then skip that OH for the rest of the script
+    for oh in ohArray:
+        if oh not in calInfo.keys():
+            calDacCalFile = "{0}/{1}/calFile_calDac_{1}.txt".format(dataPath,chamber_config[oh])
+            calDacCalFileExists = os.path.isfile(calDacCalFile)
+            if not calDacCalFileExists:
+                print("Skipping OH{0}, detector {1}, missing CFG_CAL_DAC Calibration file:\n\t{2}".format(
+                    oh,
+                    chamber_config[oh],
+                    calDacCalFile))
+                ohArray = np.delete(ohArray,(oh))
+
+    if len(ohArray) == 0:
+        print('No OHs with a calFile, exiting.')
+        exit(1)
+           
+    # Initialize nested dictionaries
+    dict_DACvsADC_Graphs = nesteddict()
+    dict_RawADCvsDAC_Graphs = nesteddict()
+    dict_DACvsADC_Funcs = nesteddict()
+
+    # Initialize a TGraphErrors for each vfat
+    for oh in ohArray:
+        for vfat in range(0,24):
+             dict_DACvsADC_Graphs[oh][vfat] = r.TGraphErrors()
+             dict_RawADCvsDAC_Graphs[oh][vfat] = r.TGraphErrors()
+             #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+             dict_DACvsADC_Graphs[oh][vfat].GetXaxis().SetTitle(nameY)
+             dict_DACvsADC_Graphs[oh][vfat].GetYaxis().SetTitle(nameX)
+             dict_RawADCvsDAC_Graphs[oh][vfat].GetXaxis().SetTitle(nameY)
+             dict_RawADCvsDAC_Graphs[oh][vfat].GetYaxis().SetTitle(nameX)
+
+    outputFiles = {}         
+             
+    for oh in ohArray:
+        if scandate == 'noscandate':
+            outputFiles[oh] = r.TFile(elogPath+"/"+chamber_config[oh]+"/"+args.outfilename,'recreate')
+        else:    
+            outputFiles[oh] = r.TFile(dataPath+"/"+chamber_config[oh]+"/dacScans/"+scandate+"/"+args.outfilename,'recreate')
+
+    # Loop over events in the tree and fill plots
+    for event in dacScanFile.dacScanTree:
+        oh = event.link
+        vfat = event.vfatN
+
+        calibrated_ADC_value=calInfo[oh]['slope'][vfat]*event.dacValY+calInfo[oh]['intercept'][vfat]
+        calibrated_ADC_error=calInfo[oh]['slope'][vfat]*event.dacValY_Err+calInfo[oh]['intercept'][vfat]
```

Comment:
For calculating `dacValY_Err` in terms of voltage instead of ADC we don't want to use the intercept.  The error here is always the least sensitive bit (LSB), so this is an ADC value of 1, e.g. the ADC bin width. Applying the intercept will not give that (indeed using the sample calibration coef's I provided you get a hugely negative number).

Reply:
fixed